kolibri/content/api.py

```diff
@@ -340,3 +347,90 @@ class FileViewset(viewsets.ReadOnlyModelViewSet):
 
     def get_queryset(self):
         return models.File.objects.all()
+
+
+class RemoteChannelViewSet(viewsets.ViewSet):
+    permissions_classes = (CanManageContent,)
+
+    http_method_names = ['get']
+
+    def _cache_kolibri_studio_channel_request(self, identifier=None):
+        cache_key = get_channel_lookup_url(identifier=identifier)
+
+        # cache channel lookup values
+        if cache.get(cache_key):
+            return Response(cache.get(cache_key))
+
+        resp = requests.get(cache_key)
+
+        # always check response code of request and set cache
+        if resp.status_code == 404:
+            raise Http404(
+                _("The requested channel does not exist on the content server")
+            )
+        cache.set(cache_key, resp.json(), 60 * 10)
+
+        kolibri_mapped_response = []
+        for channel in resp.json():
+            kolibri_mapped_response.append(self._studio_response_to_kolibri_response(channel))
+
+        cache.set(cache_key, kolibri_mapped_response, 60 * 10)
+
+        return Response(kolibri_mapped_response)
+
+    @staticmethod
+    def _get_lang_native_name(code):
+        try:
+            lang_name = languages.getlang(code).native_name
+        except AttributeError:
+            logger.warning("Did not find language code {} in our le_utils.constants!".format(code))
+            lang_name = None
+
+        return lang_name
+
+
```

Comment:
flake8 warning here: only one line between methods.

Reply:
Seems to be the cause of the test failure too.